### PR TITLE
try harder to find a suitable network interface (bsc #1068336)

### PR DESCRIPTION
**Pull request by wfeldt**
After checking network cards, go through the remaining network interfaces.

`sort_a_bit()` ensures we try real hardware first and don't try anything twice.

### CODE REVIEW

File: url.c [hunk 1]
```diff
@@ -2949,31 +3005,59 @@ hd_t *sort_a_bit(hd_t *hd_list)
...
+    /* 3. wlan cards last */
 
     for(u = 0, hd = hd_list; hd; hd = hd->next) {
       if(!hd->is.wlan) hd_array[u++] = hd;
     }
     for(hd = hd_list; hd; hd = hd->next) {
       if(hd->is.wlan) hd_array[u++] = hd;
     }
-    hd_array[hds] = NULL;
-    for(u = 0; u < hds; u++) hd_array[u]->next = hd_array[u + 1];
```
Comment (jreidinger) [inline url.c]: :+1: for abstracting this relink

File: url.c [hunk 2]
```diff
@@ -1455,23 +1458,23 @@ int url_mount(url_t *url, char *dir, int (*test_func)(url_t *))
...
         break;
 
       case inst_floppy:
-        hw_item = hw_floppy;
+        hw_items = (hd_hw_item_t[]) { hw_floppy, 0 };
         break;
 
       default:
-        hw_item = hw_block;
+        hw_items = (hd_hw_item_t[]) { hw_block, 0 };
```
Comment (jreidinger) [inline url.c]: NP: I see documentation that array have to be NULL ended, so maybe it makes sense to use constant here like `{ hw_block, NULL}` ?

Comment (wfeldt) [reply url.c]: No, that's two different things. Here it's  just an array of enums, what you probably mean is the array of (hd_t *) (the detected  hardware structs).

Comment (jreidinger) [reply url.c]: ah, now I see it. It is structure for hd_list2 method. Then fine for me.

**Pull request merged**